components/date-time/time.js

```diff
@@ -10,7 +10,7 @@ import moment from 'moment';
 import { __ } from '@wordpress/i18n';
 import Button from '../button';
 
-export function TimePicker( { currentTime, onChange, is12Hour, ...args } ) {
```

Comment:
This change makes me wonder if the extra args are really necessary for the `DateTimePicker` and `DatePicker`. It may be better if we're explicit here. (for example, explicitly pass the `locale` prop to the `DatePicker`

Reply:
`TimePicker` makes 100% sense to remove the extra args from since no other variables are utilized.

`DatePicker`, and therefore `DateTimePicker`, however I believe should allow it as it will allow developers extending it to utilize everything the `DatePicker` component has to offer: filtering which dates are including, highlighting dates, `onSelect` events, enabling month/year dropdowns, etc.

Reply:
In this case, let's remove passing the extra args to the `TimePicker`